microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/health/java/MicroProfileHealthDiagnosticsParticipant.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.health.java;
+
+import static com.redhat.microprofile.jdt.internal.health.MicroProfileHealthConstants.HEALTH_ANNOTATION;
+import static com.redhat.microprofile.jdt.internal.health.MicroProfileHealthConstants.HEALTH_CHECK_INTERFACE_NAME;
+import static com.redhat.microprofile.jdt.internal.health.MicroProfileHealthConstants.LIVENESS_ANNOTATION;
+import static com.redhat.microprofile.jdt.internal.health.MicroProfileHealthConstants.READINESS_ANNOTATION;
+
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.ITypeHierarchy;
+import org.eclipse.jdt.core.ITypeRoot;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.lsp4j.Range;
+
+import com.redhat.microprofile.commons.DocumentFormat;
+import com.redhat.microprofile.jdt.core.java.IJavaDiagnosticsParticipant;
+import com.redhat.microprofile.jdt.core.java.JavaDiagnosticsContext;
+import com.redhat.microprofile.jdt.core.utils.AnnotationUtils;
+import com.redhat.microprofile.jdt.core.utils.IJDTUtils;
+import com.redhat.microprofile.jdt.core.utils.PositionUtils;
+import com.redhat.microprofile.jdt.internal.health.MicroProfileHealthConstants;
+
+/**
+ *
+ * MicroProfile Health Diagnostics:
+ * 
+ * <ul>
+ * <li>Diagnostic 1:display Health annotation diagnostic message if
+ * Health/Liveness/Readiness annotation exists but HealthCheck interface is not
+ * implemented</li>
+ * <li>Diagnostic 2: display HealthCheck diagnostic message if HealthCheck
+ * interface is implemented but Health/Liveness/Readiness annotation does not
+ * exist</li>
+ * 
+ * </ul>
+ * 
+ * <p>
+ * Those rules comes from
+ * https://github.com/MicroShed/microprofile-language-server/blob/8f3401852d2b82310f49cd41ec043f5b541944a9/src/main/java/com/microprofile/lsp/internal/diagnostic/MicroProfileDiagnostic.java#L250
+ * </p>
+ * 
+ * @author Angelo ZERR
+ * 
+ * @See https://github.com/eclipse/microprofile-health
+ *
+ */
+public class MicroProfileHealthDiagnosticsParticipant implements IJavaDiagnosticsParticipant {
+
+	@Override

```

Comment:
Do we need to override `isAvailable()` like you did with MicroProfileRestClientDiagnosticsParticipant?

Reply:
Good catch. fixed and I renamed the method to isAdpatedForDiagnostics, because we will have that too for hover feature like hover, codelens. Please review my javadoc on IJavaDiagnosticsParticipant  and tell me if it's well explained.